### DISCUSSION TITLE
Can't get answer to queries

### DISCUSSION THREAD

**Discussion by TomF8COD** [richonguzman/LoRa_APRS_iGate#207] [Q&A] [answered]
Hi Ricardo and all,
I'm wondering how the queries are working. I have an igate F8COD-10 here and a tracker F8COD-7. I'm connected to the tracker with aprsdroid in bluetooth tnc2 mode.
When I send a query like ?APRS? to my igate (or another one for that matter), I can see my msg has been aknowledged but get nothing in return. With ?APRS? I expected to get an answer from the igate with the list of queries it understands but I got nothing. Am I missing something ?
The messages are working from aprsdroid, I clicked by mistake on aprsthursday on the tracker's menu so it sent a message and I got an answer from an OM, I managed to chat with him from aprsdroid without problem, so I think my issue is related with the queries. Does anyone have any idea about it ?
73 de Tom F8COD

**Comment by richonguzman** [comment]
Dos you read the queries wiki ?

It you message to the igate any of them it should answer besides sending the ack?

**Reply by TomF8COD** [reply]
Hi Ricardo,
Thanks a lot for answering me, I've read the queries wiki and I'm wondering if I missed something because everything seems ok, I'm working from aprsdroid and for example I did send the message "?APRS?" without quotes to my igate F8COD-10 I can see the message has been acknowledged by the igate but it doesn't reply to me with the list of queries it can understand.
So I was wondering if I misunderstand something because it seemed pretty straight forward but I can't figure out why I can't make it work...

**Reply by richonguzman** [reply]
![image](https://github.com/user-attachments/assets/64031f65-61a3-47fa-b1b5-f549d63c2d95)

**Comment by richonguzman** [comment]
Can you check on the aprs.fi webpage , on the message info, the messages recorded for this station?

Maybe it was answered ?




Saludos,

Ricardo / Richon
(Enviado desde mi celular a tubos)

El 23-12-2024, a las 6:48 a. m., TomF8COD ***@***.***> escribió:

﻿

Hi Ricardo,
Thanks a lot for answering me, I've read the queries wiki and I'm wondering if I missed something because everything seems ok, I'm working from aprsdroid and for example I did send the message "?APRS?" without quotes to my igate F8COD-10 I can see the message has been acknowledged by the igate but it doesn't reply to me with the list of queries it can understand.
So I was wondering if I misunderstand something because it seemed pretty straight forward but I can't figure out why I can't make it work...

**Comment by TomF8COD** [comment]
And with you it works. The igate did recognize you as it's master, I'm not worth of its answers :-D
Well, it works for you but I can't see my messages as it was probably rf only from F8COD-7 to and from F8COD-10. 
I've turned on a sdr to watch what's going on rf wise, I can see the tracker's message then the Igate sending one packet, witch the trackers display as 'ack' on its display, then the Igate is sending ANOTHER packet witch is completely ignored by my tracker and aprs droid.
I'm goind to install the tracker fw on another Lora32 card and turn it into packet decoder mode to figure out why this packet is ignored, I'll write here what I found.

**Comment by TomF8COD** [comment]
Well, it turns out the packets that where ignored are just ack : the ?APRS? query was sent twice from aprsdroid, so the igate sent two ack. I decoded the trafic with sdrangel to be sure of what happens, and still no query answer.  

>19:46:52 12 S:37.1 SN:80.2 HF:ok HC:ok EOM:too early
>19:47:30 12 S:25.8 SN:69.2 HF:ok HC:ok FEC:ok CRC:ok
000|3c ff 01 46|38 43 4f 44|2d 37 3e 41|50 44 52 31
016|36 2c 57 49|44 45 31 2d|31 3a 3a 46|38 43 4f 44
032|2d 31 30 20|3a 20 3f 41|50 52 53 56|7b 31 36 0a
048|4a bf 
TXT|<�F8COD-7>APDR16,WIDE1-1::F8COD-10 : ?APRSV{16

>19:47:36 12 S:37.0 SN:80.2 HF:ok HC:ok FEC:ok CRC:ok
000|3c ff 01 46|38 43 4f 44|2d 31 30 3e|41 50 4c 52
016|47 31 2c 52|46 4f 4e 4c|59 2c 57 49|44 45 32 2d
032|32 3a 3a 46|38 43 4f 44|2d 37 20 20|3a 61 63 6b
048|31 36 41 c7|
TXT|<�F8COD-10>APLRG1,RFONLY,WIDE2-2::F8COD-7  :ack16
>19:47:43 12 S:36.9 SN:80.4 HF:ok HC:ok FEC:ok CRC:ok
000|3c ff 01 46|38 43 4f 44|2d 31 30 3e|41 50 4c 52
016|47 31 2c 52|46 4f 4e 4c|59 2c 57 49|44 45 32 2d
032|32 3a 3a 46|38 43 4f 44|2d 37 20 20|3a 61 63 6b
048|31 36 41 c7|
TXT|<�F8COD-10>APLRG1,RFONLY,WIDE2-2::F8COD-7  :ack16

I tried to send a message by using another lora32 with the igate firmware, I used it as a simple serial kiss tnc with Xastir on my laptop, there it worked, F8COD-10 answered with the list of queries : 

>21:13:22 12 S:25.1 SN:68.9 HF:ok HC:ok FEC:ok CRC:ok
000|3c ff 01 46|38 43 4f 44|2d 31 3e 41|50 58 32 32
016|30 2c 57 49|44 45 31 2d|31 3a 3a 46|38 43 4f 44
032|2d 31 30 20|3a 3f 41 50|52 53 3f 7b|30 42 7d 0d
048|e6 80 
TXT|<�F8COD-1>APX220,WIDE1-1::F8COD-10 :?APRS?{0B}

>21:13:29 12 S:36.8 SN:80.2 HF:ok HC:ok FEC:ok CRC:ok
000|3c ff 01 46|38 43 4f 44|2d 31 30 3e|41 50 4c 52
016|47 31 2c 52|46 4f 4e 4c|59 2c 57 49|44 45 32 2d
032|32 3a 3a 46|38 43 4f 44|2d 31 20 20|3a 3f 41 50
048|52 53 56 20|3f 41 50 52|53 50 20 3f|41 50 52 53
064|4c 20 3f 41|50 52 53 48|20 3f 57 48|45 52 45 20
080|63 61 6c 6c|73 69 67 6e|8b aa 
TXT|<�F8COD-10>APLRG1,RFONLY,WIDE2-2::F8COD-1  :?APRSV ?APRSP ?APRSL ?APRSH ?WHERE callsign

So my problem is located between the tracker and Aprsdroid, I'm going to install the tracker on the Lora32 module and use it connected via bluetooth with Xastir on my laptop.

**Reply by iw2ejh** [reply]
I run a test on my two igates (IW2EJH-11 & IW2EJH-13) (one connected to internet and one only rf ) and I noticed the one connected to APRS.NET is responding to the queries, the one only RF is not responding.

**Reply by pdevlin13** [reply]
> I run a test on my two igates (IW2EJH-11 & IW2EJH-13) (one connected to internet and one only rf ) and I noticed the one connected to APRS.NET is responding to the queries, the one only RF is not responding.

I also see this, if I have an igate connected to the internet I get a query response but no response if connected RF only, when in digi backup

**Reply by richonguzman** [reply]
Maybe I added a RFONLY in packet when answering RF queries; that avoids answering on rf on certain cases

Will check this out

**Comment by TomF8COD** [comment, marked as answer]
I can't believe it guys.
I just realised that my phone's keyboard is putting a space after and BEFORE when I type '?'... So instead of '?APRS?' I was sending ' ?APRS ?' with Aprsdroid. So with a leading space before the query, it wasn't recognized as a proper query, hence the 'ack' from the igate and no other answers !

**Reply by richonguzman** [reply]
Great that you find out the problem and all is working fine there ;)
